File: python/arcade-intro/08_Diving Deeper/037_arrayMaxConsecutiveSum.py
def arrayMaxConsecutiveSum(inputArray, k):
    '''
    Given array of integers, find the maximal possible sum of some of its k consecutive elements.
    '''
    maxValue = 0
    for i in range(k):
        maxValue += inputArray[i]
    
    aux = maxValue
    for i in range(k, len(inputArray)):
        aux -= inputArray[i-k]
        aux += inputArray[i]
        if aux > maxValue:
            maxValue = aux
    return maxValue    

File: arcade/go/arcade-intro/08_Diving Deeper/037_arrayMaxConsecutiveSum.go
func arrayMaxConsecutiveSum(inputArray []int, k int) int {
	currentMax := 0

	if k >= len(inputArray) {
		return 0
	}

	sum := 0
	previous := 0
	for i := range inputArray {
		if i > k-1 {
			sum -= inputArray[previous]
			previous++
		}

		sum += inputArray[i]
		if sum > currentMax {
			currentMax = sum
		}
	}
	return currentMax
}
